packages/emitter-framework/src/python/utils/operation.ts

```diff
@@ -0,0 +1,113 @@
+import { typingModule } from "#python/builtins.js";
+import { refkey, type Children, type Refkey } from "@alloy-js/core";
+import * as py from "@alloy-js/python";
+import type { Model, ModelProperty } from "@typespec/compiler";
+import { useTsp } from "../../core/index.js";
+import { Atom } from "../components/atom/atom.jsx";
+import { TypeExpression } from "../components/type-expression/type-expression.jsx";
+import { efRefkey } from "./refkey.js";
+
+export interface BuildParameterDescriptorsOptions {
+  params?: (py.ParameterDescriptor | string)[];
+  mode?: "prepend" | "append" | "replace";
+  suffixRefkey?: Refkey;
+}
+
+/**
+ * Build a parameter descriptor array from a TypeSpec Model.
+ */
+export function buildParameterDescriptors(
```

Comment:
We should push for moving `typescript/utils/operation.ts` into `core` instead of duplicating all of it here.